Src/FluentAssertions/Collections/CollectionAssertions.cs

```diff
@@ -413,15 +413,13 @@ public AndConstraint<TAssertions> BeEquivalentTo(IEnumerable expectation, string
 
             EquivalencyAssertionOptions<IEnumerable> options = config(AssertionOptions.CloneDefaults<IEnumerable>());
 
-            var context = new EquivalencyValidationContext
+            var context = new EquivalencyValidationContext(Node.From<IEnumerable>(CallerIdentifier.DetermineCallerIdentity()))
```

Comment:
Without having done any benchmarking to back this claim up, it sounds expensive to eagerly call `CallerIdentifier.DetermineCallerIdentity`.

Reply:
Made it the evaluation deferred .